packages/visual-editor/src/hooks/useReadOnlyComponent.tsx

```diff
@@ -0,0 +1,116 @@
+import { RenderDropzoneFunction } from '@components/DraggableBlock/Dropzone.types';
+import { isContentfulStructureComponent } from '@contentful/experiences-core';
+import { MissingComponentPlacehoder } from '@components/DraggableHelpers/MissingComponentPlaceholder';
+
+type UseComponentProps = {
+  node: ExperienceTreeNode;
+  resolveDesignValue: ResolveDesignValueType;
+  renderDropzone: RenderDropzoneFunction;
+};
+
+export const useReadOnlyComponent = ({
+  node: rawNode,
+  resolveDesignValue,
+  renderDropzone,
+}: UseComponentProps) => {
+  const areEntitiesFetched = useEntityStore((state) => state.areEntitiesFetched);
+  const entityStore = useEntityStore((state) => state.entityStore);
+
+  const node = useMemo(() => {
+    if (rawNode.type === ASSEMBLY_NODE_TYPE && areEntitiesFetched) {
+      return resolveAssembly({
+        node: rawNode,
+        entityStore,
+      });
+    }
+
+    return rawNode;
+  }, [areEntitiesFetched, rawNode, entityStore]);
+
+  const componentRegistration: ComponentRegistration | undefined = useMemo(() => {
+    let registration = componentRegistry.get(node.data.blockId!);
+
+    if (node.type === ASSEMBLY_NODE_TYPE && !registration) {
+      registration = createAssemblyRegistration({
+        definitionId: node.data.blockId!,
+        component: Assembly,
+      });
+    }
+
+    if (!registration) {
+      console.warn(
+        `Component registration not found for component with id: "${node.data.blockId}". The component might of been removed. To proceed, remove the component manually from the layers tab.`,
+      );
+      return undefined;
+    }
+    return registration;
+  }, [node]);
+
+  const componentId = node.data.id;
+
+  const { componentProps, sizeStyles } = useComponentProps({
+    node,
+    areEntitiesFetched,
+    resolveDesignValue,
+    renderDropzone,
+    definition: componentRegistration?.definition,
+  });
+
+  const elementToRender = (props) => {
+    if (!componentRegistration) {
+      return <MissingComponentPlacehoder blockId={node.data.blockId} />;
+    }
+
+    const { editorMode: _editorMode, node: _node, ...customComponentProps } = componentProps;
+
+    const isStructureComponent = isContentfulStructureComponent(node.data.blockId);
+    const isAssembly = node.type === 'assembly';
+    const modifiedProps =
+      isStructureComponent || isAssembly ? componentProps : customComponentProps;
+
+    const requiresDragWrapper =
+      !isStructureComponent && componentRegistration.options?.wrapComponent === false;
+
+    const { children, innerRef, style, ...rest } = props;
+
+    const element = React.createElement(
+      ImportedComponentErrorBoundary,
+      null,
+      React.createElement(componentRegistration.component, {
+        ...modifiedProps,
+        style: { ...sizeStyles, ...style },
+        // forces the container elements to render without the dropzone
+        editorMode: false,
```

Comment:
[7] Creating the element that will be rendered for this node. We set it here directly to editor mode as this would prevent elements like the container to not render the dropzone